### PR TITLE
NAS-122335 / 23.10 / Add `jenkins` to `community` train

### PR DESCRIPTION
**Pull request by stavros-k**
- Runs rootless (UID not configurable)
- Can use certificate for HTTPS
- Additional storage/volumes
- Additional env variables
- Additional Java Jenkins Options / Jenkins Options.
- Validation for duplicate entries / reserved options
- chown on mounted volumes only if parent dir has a mismatch on user/group

### CODE REVIEW

File: library/ix-dev/community/jenkins/templates/_validation.tpl [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{{- define "jenkins.validation" -}}
+  {{- if and (not .Values.jenkinsNetwork.http) (not .Values.jenkinsNetwork.https) -}}
+    {{- fail "Jenkins - At least one of [HTTP, HTTPS] must be enabled." -}}
```
Comment (Qubad786) [inline library/ix-dev/community/jenkins/templates/_validation.tpl:3]: Can both be enabled at the same time?

Comment (Qubad786) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: Just to be sure both http and https work when both options are checked?

Comment (stavros-k) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: Yes, that's correct

Comment (Qubad786) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: In practice it is better to not allow http access when https is enabled, is there specific motivation for this setup?

Comment (stavros-k) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: I agree, only reason I've "allowed" it is if there are any other services that need to interact with Jenkins and do not like https.

Comment (Qubad786) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: Are there any such services at the moment?

Comment (stavros-k) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: I'm not aware of any.

Comment (Qubad786) [reply library/ix-dev/community/jenkins/templates/_validation.tpl:3]: @stavros-k let's remove this behaviour in that case then ;)

**Review by Qubad786 — APPROVED** [review]

**Pull request merged**